tasks/pipeline.py

```diff
@@ -856,24 +856,26 @@ def compare_to_itself(ctx):
     for attempt in range(max_attempts):
         print(f"[{datetime.now()}] Waiting 30s for the pipelines to be created {attempt + 1}/{max_attempts}")
         time.sleep(30)
-        pipelines = agent.pipelines.list(ref=new_branch, get_all=True)
-        for pipeline in pipelines:
+        for pipeline in agent.pipelines.list(ref=new_branch, get_all=True):
```

Comment:
Unrelated, but I'd however recommend `iterator=True` over `get_all=True` esp. when there's no need to have the full list of pipelines (potentially large) in memory.
```suggestion
        for pipeline in agent.pipelines.list(ref=new_branch, iterator=True):
```
(there are several occurrences in the present file)